File: CppPractice/Project1/Project1/Leetcode1480.cpp
#include <string>

#include <vector>

#include <iostream>

#include <stdlib.h>

#include <time.h>

#include <sstream>

#include <algorithm>

using namespace std;


class Solution1480 {
public:
    vector<int> runningSum(vector<int>& nums) {
        vector<int> ans;
        int temp_sum = 0;
        for (int i = 0; i < nums.size(); i++) {
            temp_sum = temp_sum + nums[i];
            ans.push_back(temp_sum);
        }
        return ans;


    }
};

File: Pythonleetcode/Leetcode1480.py
class Solution(object):
    def runningSum(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        ans=[0]*len(nums)
        ans[0]=nums[0]
        for i in range(1,len(nums)):
            ans[i] = ans[i-1]+nums[i]

        return ans


print(Solution().runningSum([1,2,3,4]))
print(Solution().runningSum( [1,1,1,1,1]))
print(Solution().runningSum([3,1,2,10,1]))
